### PR TITLE
feat: find best_detections for classifications (#631)

### LINKED ISSUE #631: Support best detections based on classification_id, stream_id and both
**Is your feature request related to a problem? Please describe.**
A user sometimes wants to select best detections based on classification_id, and sometimes may include both (and best detections by day as well)

**Describe the solution you'd like**
Add more columns to the best_detections table:
classification_id
stream_ranking, // exists
stream_daily_ranking, // now is daily_ranking
classification_ranking, // to add
classification_daily_ranking, // add
stream_classification_ranking, // add
stream_classification_daily_ranking, // add

**Additional context**

### PR DESCRIPTION
**Pull request by veckatimest**
## ✅ DoD

- [x] Resolves #631 
- [ ] API docs updated na
- [ ] Release notes updated na
- [ ] Deployment notes updated na
- [ ] Unit or integration tests added na
- [ ] DB migrations tested na

_(use na when API docs (Release notes, etc) do not need to be updated)_

## 📝 Summary

Now when "classifications" query parameter is passed to 
GET /classifier-jobs/{jobId}/best-detections
or
GET /classifier-jobs/{jobId}/best-detections/summary
The returned detections will be related to specific classification_id

## 🛑 Problems

- Write any discovered & unresolved problems (link to created Jira issues)

## 💡 More ideas

Write any more ideas you have

### CODE REVIEW

File: core/_cli/migrations/20240404131807-create-best-detections.ts.js [hunk 1]
```diff
@@ -56,46 +56,47 @@ module.exports = {
...
+      //         FROM public.detections
+      //         WHERE (start BETWEEN :jobStart AND :jobEnd) AND classifier_job_id = :classifierJobId
+      //       ) as detection
+      //     WHERE daily_ranking < :perDayLimit OR stream_ranking < :perStreamLimit;
+      //   `, {
+      //     replacements,
+      //     type: Sequelize.QueryTypes.RAW,
+      //     transaction: t
+      //   })
+      // }
```
Comment (Tooseriuz) [inline core/_cli/migrations/20240404131807-create-best-detections.ts.js:99]: better just remove it if not needed

File: core/detections/best-detections-summary.int.test.js [hunk 2]
```diff
@@ -123,21 +118,24 @@ let job2Stream1Day2BestDetection
...
-        arbitraryDetections.push(oneDetection({
-          streamId: stream.id,
-          start: new Date(date),
-          end: new Date(date + 5000),
-          confidence: 0.7 + Math.random() / 10
-        }))
-
-        date += 60 * 60 * 1000
+    classifications.forEach((classification) => {
+      // we want 6 days of detections, 20 detections each day
```
Comment (Tooseriuz) [inline core/detections/best-detections-summary.int.test.js:122]:
```suggestion
      // we want 6 days of detections, 10 detections each day
```

File: core/detections/best-detections.js [hunk 3]
```diff
@@ -22,6 +22,15 @@ const Converter = require('../../common/converter')
  *         description: List of stream ids to limit results
  *         in: query
  *         type: array|string
+ *       - name: classifications
```
Comment (Tooseriuz) [inline core/detections/best-detections.js:25]:
```suggestion
 *       - name: classificationIds
```
some other places, we use `classifications` for classification name not ids. so better make it consistent

File: core/detections/best-detections.js [hunk 4]
```diff
@@ -83,18 +92,20 @@ router.get('/:jobId/best-detections', (req, res) => {
   const { jobId } = req.params
   const converter = new Converter(req.query, {}, true)
   converter.convert('streams').optional().toArray()
+  converter.convert('classifications').optional().toArray()
```
Comment (Tooseriuz) [inline core/detections/best-detections.js:95]:
```suggestion
  converter.convert('classification_ids').optional().toArray()
```

File: core/detections/dao/index.js [hunk 5]
```diff
@@ -90,7 +90,7 @@ async function defaultQueryOptions (filters = {}, options = {}) {
 
 async function defaultBestDetectionsQueryOptions (filters, opts) {
   const { user, limit, offset, fields } = opts
-  const { streams, reviewStatuses } = filters
+  const { streams, classifications, reviewStatuses } = filters
```
Comment (Tooseriuz) [inline core/detections/dao/index.js:93]:
```suggestion
  const { streams, classificationIds, reviewStatuses } = filters
```

File: core/detections/dao/index.js [hunk 6]
```diff
@@ -120,6 +120,13 @@ async function defaultBestDetectionsQueryOptions (filters, opts) {
     bestDetectionsWhere.streamId = streamIds
   }
 
+  let byClassicication = false
+  if (classifications) {
+    byClassicication = true
+
+    bestDetectionsWhere.classificationId = classifications
```
Comment (Tooseriuz) [inline core/detections/dao/index.js:127]:
```suggestion
  if (classificationIds) {
    byClassicication = true

    bestDetectionsWhere.classificationId = classificationIds
```

File: core/_cli/migrations/20240404131807-create-best-detections.ts.js
Comment (grindarius) [inline core/_cli/migrations/20240404131807-create-best-detections.ts.js:1]: Would this migration gets run again?

Comment (veckatimest) [reply core/_cli/migrations/20240404131807-create-best-detections.ts.js:1]: only somewhere where we did not have it yet (probably not)

File: core/detections/dao/index.js (hunk 6 above)
Comment (veckatimest) [reply core/detections/dao/index.js]: fixed

File: core/detections/dao/index.js (hunk 5 above)
Comment (veckatimest) [reply core/detections/dao/index.js]: fixed

**Review by Tooseriuz — APPROVED** [review]

**Pull request merged**